src/settings.rs

```diff
@@ -0,0 +1,43 @@
+use crate::connection::Connection;
+use crate::dbus_api::DBusAccessor;
+use crate::errors::Error;
+use crate::gen::OrgFreedesktopNetworkManagerSettings;
+use dbus::blocking::Connection as DBusConnection;
+
+const SETTINGS_BUS: &str = "org.freedesktop.NetworkManager.Settings";
+const SETTINGS_PATH: &str = "/org/freedesktop/NetworkManager/Settings";
+
+pub struct Settings<'a> {
+    dbus_accessor: DBusAccessor<'a>,
+}
+
+impl<'a> Settings<'a> {
+    pub fn new(dbus_connection: &'a DBusConnection) -> Self {
+        Self {
+            dbus_accessor: DBusAccessor::new(dbus_connection, SETTINGS_BUS, SETTINGS_PATH),
+        }
+    }
+
+    pub fn list_connections(&self) -> Result<impl Iterator<Item = Connection>, Error> {
```

Comment:
Hi @Xaeroxe, 

It seems that the main approach is a vector in other parts of the code. In this case, Could you tell me more about the usage of an `impl Iterator` VS a vector?

B. r

Reply:
Sure! An Iterator is a structure which describes how to obtain the data contained within. Whereas a Vec is a heap allocated collection containing the data. An Iterator can be turned into a Vec by collecting it, and a Vec can be iterated in several ways. The Iterator can be interacted with without requiring a heap allocation, and can be combined with other Iterators. In the end the two approaches are very similar but an Iterator might be more efficient.